### PR TITLE
`vmmap -A / -B` improvements

### LINKED ISSUE #1821: vmmap -Ax -Bx displays wrong results when filter returns multiple pages
image

### PR DESCRIPTION
**Pull request by feelfreelinux**
Implements improvements over https://github.com/pwndbg/pwndbg/pull/1810
- `vmmap -A / -B` now handles multiple results gracefully
- printed pages now prepend an backtrace arrow before the results that were actually filtered, to make it more readable
- Fixed displaying of address offset in filtered pages

Fixes: https://github.com/pwndbg/pwndbg/issues/1821 and https://github.com/pwndbg/pwndbg/issues/1766

### CODE REVIEW

File: pwndbg/commands/vmmap.py [hunk 1]
```diff
@@ -83,42 +83,72 @@ def print_vmmap_table_header() -> None:
...
-                    filtered_pages.append(pages[matched_index + after_index])
+                if (
+                    matched_index + after_index < len(total_pages) - 1
+                    and total_pages[matched_index + after_index] not in pages_to_display
+                ):
+                    pages_to_display.append(total_pages[matched_index + after_index])
 
-        pages = filtered_pages
+        # Sort results by address
+        total_pages = sorted(pages_to_display, key=lambda page: page.vaddr)
```
Comment (disconnect3d) [inline pwndbg/commands/vmmap.py:127]: Hmm aren't they sorted already from vmmap.get?

File: pwndbg/color/memory.py [hunk 2]
```diff
@@ -16,14 +16,15 @@
...
-def get(address, text=None) -> str:
+def get(address, text=None, prefix=None) -> str:
     """
     Returns a colorized string representing the provided address.
 
     Arguments:
         address(int): Address to look up
         text(str): Optional text to use in place of the address
               in the return value string.
+        prefix(str): Optional text to set at beginning in the return value string.
```
Comment (disconnect3d) [inline pwndbg/color/memory.py:27]: Hmm shall we add the prefix here or just format it inline when used? (cc: @gsingh93 wdyt?)

File: pwndbg/commands/vmmap.py (hunk 1 above)
Comment (feelfreelinux) [reply pwndbg/commands/vmmap.py:127]: I sorted it at the end to simplify the insertion logic of the look around pages a bit, as it was a little bit messy after adding the logic to insert pages in-between multiple results.

File: pwndbg/color/memory.py (hunk 2 above)
Comment (gsingh93) [reply pwndbg/color/memory.py:27]: Yea, I feel like the prefix should be handled outside of this function.

Comment (disconnect3d) [reply pwndbg/color/memory.py:27]: @feelfreelinux any chance u can revert that small change and handle it outside?

**Pull request merged**